Cython/Compiler/Main.py

```diff
@@ -502,6 +502,9 @@ def run_pipeline(source, options, full_module_name=None, context=None):
 
     err, enddata = Pipeline.run_pipeline(pipeline, source)
     context.teardown_errors(err, options, result)
+    if options.depfile:
+        from ..Build.Dependencies import DependencyTree
+        DependencyTree(context).write_depfile_for(result.main_source_file, result.c_file)
```

Comment:
There's a function `create_dependency_tree()` as entry point.

Reply:
Okay. I have no idea what the context is at different points, or whether the cached DependencyTree makes a difference, but if you say to use it, I'll use it. :)